core/src/main/java/io/substrait/extendedexpression/ExtendedExpression.java

```diff
@@ -0,0 +1,36 @@
+package io.substrait.extendedexpression;
+
+import io.substrait.expression.Expression;
+import io.substrait.proto.AdvancedExtension;
+import io.substrait.relation.Aggregate;
+import io.substrait.type.NamedStruct;
+import java.util.List;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class ExtendedExpression {
+  public abstract List<ExpressionReferenceBase> getReferredExpressions();
+
+  public abstract NamedStruct getBaseSchema();
+
+  public abstract List<String> getExpectedTypeUrls();
+
+  // creating simple extensions, such as extensionURIs and extensions, is performed on the fly
+
+  public abstract Optional<AdvancedExtension> getAdvancedExtension();
```

Comment:
This needs to be the POJO AdvancedExtension not the proto one.

Reply:
Wait, the Plan POJO also use the protobuf AdvancedExtension.

https://github.com/substrait-io/substrait-java/blob/e73c6cac7a5b62d026ccd13eca97bfae02395aab/core/src/main/java/io/substrait/plan/Plan.java#L3C2-L3C45

I'll file an issue to fix this in the future.

Reply:
Filed: https://github.com/substrait-io/substrait-java/issues/210